src/collections/files.ts

```diff
@@ -1,20 +1,35 @@
 import { BaseCollection } from './base_collection';
 import { File } from '../models/file';
+import { QueuedProcess } from '../models/queued_process';
 import { UploadFileParams } from '../interfaces/upload_file_params';
 import { DownloadFileParams } from '../interfaces/download_file_params';
 
 export class Files extends BaseCollection {
-  protected static rootElementName:string = 'files';
-  protected static prefixURI:string = 'projects/{!:project_id}/files/{:id}';
+  protected static rootElementName: string = 'files';
+  protected static prefixURI: string = 'projects/{!:project_id}/files/{:id}';
   protected static elementClass: Object = File;
 
-  upload(project_id: string, upload: UploadFileParams) {
-    return this.createPromise('POST', { project_id: project_id}, this.returnBareJSON,
+  protected static secondaryElementNameSingular: string = 'process';
+  protected static secondaryElementClass: Object = QueuedProcess;
+
+  upload(project_id: string, upload: UploadFileParams): Promise<any> {
+    return this.createPromise('POST', { project_id: project_id },
+      this.choosePopulator(upload),
       this.handleReject, upload, 'projects/{!:project_id}/files/upload');
   }
 
-  download(project_id: string, download: DownloadFileParams) {
+  download(project_id: string, download: DownloadFileParams): Promise<any> {
     return this.createPromise('POST', { project_id: project_id }, this.returnBareJSON,
       this.handleReject, download, 'projects/{!:project_id}/files/download');
   }
+
+  private choosePopulator(uploadParams: UploadFileParams): any {
+    // Temporary solution for backwards compatibility with sync file uploading
+    // Sync uploading will be removed by summer 2020
+    if (uploadParams.queue) {
```

Comment:
I don’t think we should give the queue option as an option for the users. We should force ‘queue=true’ for all.

Then this will be the final version, and it doesn’t make sense to give users the ability to update to this version and still use the old way.